conftest.py

```diff
@@ -185,14 +185,23 @@ def driver(
 
 
 @pytest.fixture()
-def screenshot(driver: webdriver.Firefox, opt_ci: bool):
-    def _screenshot(filename):
+def screenshot(driver: webdriver.Firefox, opt_ci: bool) -> Callable:
+    """
+    Factory fixture that returns a screenshot function.
+    """
+
+    def _screenshot(filename: str) -> str:
```

Comment:
curious as to why you have a private function defined in the function with the same name? what was the design decision behind putting everything in the same function scope?

Reply:
This is a common PyTest fixture pattern called a "function factory" or "factory fixture". It's the way you add a fixture that refers to a function itself, so that you can request that fixture and receive a callable that you can later execute with arguments.